experimental/Rings/WeierstrassPreparationTheorem.jl

```diff
@@ -0,0 +1,240 @@
+##############################################################
+#
+#Conversion from Singular (weierstr.lib)
+#
+#############################################################
+
+export weierstrass_preparation_theorem
+
+export weierstrass_division_theorem
+
+
+"""
+`order_of_generality(f::MPolyElem, n::Int)`
```

Comment:
Please use exactly four spaces to indent the signature(s) in docstrings, and use no extra backticks around it
```suggestion
    order_of_generality(f::MPolyElem, n::Int)
```